bezier-rs/lib/src/lib.rs

```diff
@@ -1,4 +1,112 @@
+use glam::DVec2;
+
 /// Test function to double a number
 pub fn test_double(num: i32) -> i32 {
 	num + num
 }
+
+enum BezierType {
+	Quadratic,
+	Cubic,
+}
+
+/// Representation of a bezier curve with 2D points
+pub struct Bezier {
+	/// Vector containing the bezier points (represented as DVec2s)
+	points: [Option<DVec2>; 4],
+	/// The type of bezier curve
+	bezier_type: BezierType,
+}
```

Comment:
Would it be easier to have the points inside the enum to reduce the need to unwrap when using the points?
```rs
pub enum BezierSegement {
    Quadratic { start: DVec2, handle: DVec2, end: DVec2 },
    Cubic { start: DVec2, handle1: DVec2, handle2: DVec2, end: DVec2 },
}
```
This would additionally reduce the memory from 104 bytes to 72.

Reply:
We refactored the code using this idea - do you have any thoughts on the new approach?

Reply:
Looks good - it is nice to not have to have unwrap all over the code.